monitor/src/main.rs

```diff
@@ -5,18 +5,16 @@
 // license that can be found in the LICENSE file or at
 // https://opensource.org/licenses/MIT.
 
-#![feature(lang_items)]
-#![feature(start)]
 #![feature(strict_provenance)]
 #![cfg_attr(not(test), no_main)]
 #![cfg_attr(not(test), no_std)]
+#![forbid(absolute_paths_not_starting_with_crate)]
+#![forbid(elided_lifetimes_in_paths)]
 #![forbid(unsafe_op_in_unsafe_fn)]
 
-libhypatia::define_segment!(init);
-
-// XXX(mikew): For some reason, removing this no_mangle on this init in particular causes
```

Comment:
FWIW, what I suspect happened here is that the entry point in the ELF header was 0, and Theon happy jumped off into nothingness, likely causing an exception that resulted in a panic!(), but most panic!() handlers don't print anything, so the machine just halted. At least, that's my guess.

Reply:
Actually, it's even worse than that. Indeed, the ELF entry point is zero, but the exception handler actually returns, jumping back to zero, and then reinvoking itself, forever.

Gotta do some better error handling at some point....